src/deepsparse/pipeline.py

```diff
@@ -462,6 +491,18 @@ def engine_type(self) -> str:
         """
         return self._engine_type
 
+    def in_dynamic_batch_mode(self) -> bool:
+        """
+        :return: True if pipeline should be run in dynamic batch mode else
+            False
+        """
+        return (
+            self._batch_size is None
+            and self.executor
+            and issubclass(self.input_schema, Splittable)
```

Comment:
Maybe throw an error if one but not all of these conditions is met

Reply:
the only issue might be batch size is None, executor and splittable are fine